src/OrderingComparer.ts

```diff
@@ -0,0 +1,146 @@
+import { String as S } from "./string"
+
+type CompareResult =
```

Comment:
Does this match JavaScript's built in sort? (at least, these comments explaining the different values).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

```
function compareFn(a, b) {
  if (a is less than b by some ordering criterion) {
    return -1;
  }
  if (a is greater than b by the ordering criterion) {
    return 1;
  }
  // a must be equal to b
  return 0;
}
```

I'm wondering if maybe I'm confused because before and after imply a sort _order_ (ASC or DESC) and not just "greater than or less than"?

Reply:
It does match the default sort result, I updated the comments to reflect greater than/less than instead of before/after (although I think it was legitimate to say it that way as well).

You can peep all the tests, but I'm like 99.999% sure that this matches the way JS does it.

Reply:
I like the new docs, thanks!